docs/guides/federation.md

```diff
@@ -2,6 +2,175 @@
+can create services using Strawberry and federate them via Apollo Gateway.
+
+> _NOTE_: we don’t have a gateway server, you’d need to always use the Apollo
+> Gateway for this.
+
+Apollo Federation allows to combine multiple GraphQL APIs into one. This can be
+extremely useful when working with micro-services.
+
+## Federated schema example
+
+Let’s look at an example on how to implement Apollo Federation using Strawberry,
+We’ll have an application with two federated services.
+
+1. book: a service with all the books we have
+2. reviews: a service with book reviews.
+
+### Books service
+
+```python
+@strawberry.federation.type(keys=["id"])
+class Book:
+    id: strawberry.ID
+    title: str
+
+def get_all_books() -> List[Book]:
+    return [Book(id=1, title="The Dark Tower")]
+
+@strawberry.type
+class Query:
+    all_books: List[Book] = strawberry.field(resolver=get_all_books)
+
+schema = strawberry.federation.Schema(query=Query)
+```
+
+We defined two types, `Book` and `Query`, `Query` has only one field that allows
+to fetch all the books.
+
+Meanwhile `Book` used the `strawberry.federation.type` decorator, as opposed to
+the normal `strawberry.type`, this new decorator extend the base one and allows
+to define federation specific attributes to the type.
+
+In this case we are telling federation that the key to uniquely identify a book
+is the `id` field.
+
+> Federation keys can be thought primary keys. They are used by the gateway to
+> query types between multiple services and then joining them into the augmented
+> type.
+
+### Reviews service
+
+Let’s look at how our review service looks like, we want to define a type for a
+review but also extend the book to have a list of review.
+
+```python
+@strawberry.type
+class Review:
+    body: str
+
+def get_reviews() -> List[Review]:
+    return [Review(body="This is a review")]
+
+@strawberry.federation.type(extend=True, keys=["id"])
+class Book:
+    id: strawberry.ID = strawberry.federation.field(external=True)
+    reviews: List[Review] = strawberry.field(resolver=get_reviews)
+
+    @classmethod
+    def resolve_reference(cls, id: strawberry.ID):
+        return Book(id)
+```
+
+Now things are looking more interesting, the `Review` type is a GraphQL type
+that holds the content of the review.
+
+Meanwhile we are able to extend the `Book` type by using
+`strawberry.federation.type` again and passing `extend=True` as a parameter.
+This tells federation that we are extending an already existing type.
+
+We are also declaring two fields on `Book`, one is the `id` which is marked as
+external with `strawberry.federation.field(external=True)`, this tells
+federation that this field is not available in this service, and **that** it
+comes from another service.
+
+The other field is `reviews` which results in a list of `Reviews` for this book.
+
+Finally we also have a class method called `resolve_reference` that allows us to
```

Comment:
It's bit hard to understand what the `resolve_references` is doing here. What might help is to add  another field to the Book type that gets set in the resolve_references function? Something like a `review_count` field?

Reply:
done! let me know what you think :)

Reply:
It's a shame it's so contrived but I think it does the job of showing the purpose of the function 👍